### DISCUSSION TITLE
Unable to Filter Using Combined Conditions on Related Tables in Supabase Query

### DISCUSSION THREAD

**Discussion by engnaveed** [supabase/supabase#28003] [Questions] [answered]
### Issue Description

I am experiencing an issue when trying to filter results from the `bookings` table based on multiple conditions involving both direct columns and related table columns.

### Related Discussions
I have tried various solutions, including discussions and issues from the following links: [#787](https://github.com/orgs/supabase/discussions/787),  [#6778](https://github.com/orgs/supabase/discussions/6778),  [#18997](https://github.com/orgs/supabase/discussions/18997),  [PostgREST Issue #1075](https://github.com/PostgREST/postgrest/issues/1075)

 ```javascript
   const baseQuery = supabase
     .from('bookings')
     .select(
       `id, inserted_at,
       dispatch_client_org_name:dispatch_client_orgs!inner(id, name, avatar), 
       dispatcher_name:user_profiles!dispatcher(user_id, name), dropoff_date`,
       { count: 'exact' }
     )
     .eq('dispatch_org_id', dispatch_org_id)
     .order('inserted_at', { ascending: false });

   if (searchTerm) {
     baseQuery.or(
       `load_number.ilike.%${searchTerm}%,pickup_city.ilike.%${searchTerm}%,dispatch_client_orgs.name.ilike.%${searchTerm}%`
     );
   }
   ```

#### Expected Behavior
I expect to retrieve records from the `bookings` table where:
- Either `load_number` or `pickup_city` matches the `searchTerm`, **or**
- The `dispatch_client_orgs.name` matches the `searchTerm`.

#### Actual Behavior
When attempting to combine these filters using the `or` method, I receive an error. Here’s the problematic code:

```javascript
if (searchTerm) {
  baseQuery.or(
    `load_number.ilike.%${searchTerm}%,pickup_city.ilike.%${searchTerm}%,dispatch_client_orgs.name.ilike.%${searchTerm}%`
  );
}
```

This results in the following error:

```
code: 'PGRST100',
details: 'unexpected "a" expecting "not" or operator (eq, gt, ...)',
message: '"failed to parse logic tree ((or(load_number.ilike.%alpha%,pickup_city.ilike.%alpha%),or(dispatch_client_orgs_id.name.ilike.%alpha%)))" (line 1, column 87)'

//also try to add referenceTable but no luck 
```

#### Working Conditions
1. Filtering using only `load_number` and `pickup_city` works fine:
   ```javascript
   if (searchTerm) {
     baseQuery.or(
       `load_number.ilike.%${searchTerm}%,pickup_city.ilike.%${searchTerm}%`
     );
   }
   ```

2. Filtering using only `dispatch_client_orgs.name` also works:
   ```javascript
   if (searchTerm) {
     baseQuery.ilike('dispatch_client_orgs.name', `%${searchTerm}%`);
   }
   ```

#### Steps to Reproduce
1. Set up the necessary database tables and relationships as defined below:
   ```sql

   CREATE TABLE dispatch_client_orgs (
     id SERIAL PRIMARY KEY,
     name TEXT,
     avatar TEXT
   );

   CREATE TABLE bookings (
     id SERIAL PRIMARY KEY,
     dispatch_client_org_id INTEGER REFERENCES dispatch_client_orgs(id),
     load_number TEXT,
     pickup_city TEXT,
     dispatch_org_id UUID
   );
   ```

2. Execute the following query:
const baseQuery = supabase
  .from('bookings')
  .select(
    `id,
    dispatch_client_org_name:dispatch_client_orgs!inner(id, name, avatar), `,
    { count: 'exact' }
  )
  .eq('dispatch_org_id', dispatch_org_id)
  .order('inserted_at', { ascending: false });

if (searchTerm) {
  baseQuery.or(
    `load_number.ilike.%${searchTerm}%,pickup_city.ilike.%${searchTerm}%,dispatch_client_orgs.name.ilike.%${searchTerm}%`
  );
}

3. Observe the error in the response.
```
code: 'PGRST100',
details: 'unexpected "a" expecting "not" or operator (eq, gt, ...)',
message: '"failed to parse logic tree ((or(load_number.ilike.%alpha%,pickup_city.ilike.%alpha%),or(dispatch_client_orgs_id.name.ilike.%alpha%)))" (line 1, column 87)'
```

### Additional Information
I have attempted various solutions and consulted multiple resources, including Stack Overflow and the Supabase documentation, but none have resolved the issue. Any guidance or suggestions for a fix would be greatly appreciated!

**Comment by GaryAustin1** [comment, marked as answer]
.or only works on one table.  It can be the main query table or a foreign table, but all conditions in the .or must be the same table.

**Reply by GaryAustin1** [reply]
There is a possible workaround in some cases:
https://github.com/orgs/supabase/discussions/18997

**Reply by engnaveed** [reply]
@GaryAustin1 Thanks for the reply! I already tried the workaround mentioned in [#18997 (https://github.com/orgs/supabase/discussions/18997), but unfortunately, it didn’t work for my case.

**Reply by GaryAustin1** [reply]
Then you will need to use an rpc call to a table returning function or a view.

**Reply by GaryAustin1** [reply]
Also when you do filter chaining you may need to add the modifiers (like order) to the very end of your query.

**Reply by GaryAustin1** [reply]
But .or() will not work with multiple tables, so if the linked issue did not help then it is not possible in the REST API.

**Reply by engnaveed** [reply]
> Also when you do filter chaining you may need to add the modifiers (like order) to the very end of your query.

No luck with this getting same error
```javascript
 baseQuery.or(
`or(load_number.ilike.%${searchTerm}%,pickup_city.ilike.%${searchTerm}%),or(dispatch_client_orgs_id.name.ilike.%${searchTerm}%)`

    code: 'PGRST100',
    details: 'unexpected "a" expecting "not" or operator (eq, gt, ...)',
    message: '"failed to parse logic tree ((or(load_number.ilike.%alpha%,pickup_city.ilike.%alpha%),or(dispatch_client_orgs_id.name.ilike.%alpha%)))" (line 1, column 87)'
```

**Reply by engnaveed** [reply]
> But .or() will not work with multiple tables, so if the linked issue did not help then it is not possible in the REST API.

is there any other way of doing this except the functions

**Reply by GaryAustin1** [reply]
No.  View or functions.  Or two queries and you combine the results, but I don't recommend that.
